### DISCUSSION TITLE
Add LSP support

### DISCUSSION THREAD

**Discussion by SkypLabs** [stoplightio/spectral#2425] [Ideas]
**User story.**

As a developer, I can use any source code editor or IDE supporting the [Language Server Protocol (LSP)](https://en.wikipedia.org/wiki/Language_Server_Protocol) to get feedback from Spectral without having to run the CLI tool over and over.

**Is your feature request related to a problem?**

It is frustrating to have to regularly run the CLI tool to get linting information from Spectral whereas they could be displayed directly inside the source code editor / IDE (Vim in my case).

**Describe the solution you'd like**

Adding a language server to Spectral.

This way, Spectral would become the very first OpenAPI LSP as far as I know!

**Additional context**

* https://microsoft.github.io/language-server-protocol/
* https://github.com/Microsoft/vscode-languageserver-node

**Comment by sebas2day** [comment]
Well there is a vscode extension for spectral that I believe implements the LSP protocol

https://github.com/stoplightio/vscode-spectral/tree/master/server

**Comment by SkypLabs** [comment]
Oh interesting! I apparently missed it.

What do you think about the idea of integrating the language server directly into Spectral? This way, any text editor / IDE supporting LSP could profit from it. And given that the hard work has already been done for the vscode extension, integrating the language server into Spectral should be straightforward.

**Comment by luizcorreia** [comment]
@sebas2day that extension was deprecated, using old version from spectral, I compiled that to use with neovim. But I want a version with new one.

**Comment by sebas2day** [comment]
@luizcorreia I have no idea, I'm not a maintainer.

**Comment by inulty** [comment]
Showing support for this enhancement 👍

**Comment by SkypLabs** [comment]
The [original issue](https://github.com/stoplightio/spectral/issues/1607) has been turned into this discussion. Is it your way @mnaumanali94 to say that you're not planning on working on this feature request?
